File: Demos/VB.Net/DataFromArray/Form1.vb
Public Class Form1
    Private FArray As Integer()

    Private Sub CreateArray()
        Me.FArray = New Integer(9) {}
        Dim i As Integer
        For i = 0 To 9
            Me.FArray(i) = (i + 1)
        Next i
    End Sub

    Private Sub Form1_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        CreateArray()
    End Sub

    Private Sub btnCreateNew_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCreateNew.Click
        ' create report instance
        Dim report As New Report
        ' register the array
        report.RegisterData(Me.FArray, "Array")
        ' design the report
        report.Design()
        ' free resources used by report
        report.Dispose()
    End Sub

    Private Sub btnRunExisting_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnRunExisting.Click
        ' create report instance
        Dim report As New Report
        ' load the existing report
        report.Load("..\..\report.frx")
        ' register the array
        report.RegisterData(Me.FArray, "Array")
        ' run the report
        report.Show()
        ' free resources used by report
        report.Dispose()
    End Sub
End Class

File: Demos/C#/DataFromArray/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using FastReport;

namespace DataFromArray
{
  public partial class Form1 : Form
  {
    private int[] FArray;
    
    public Form1()
    {
      InitializeComponent();
      CreateArray();
    }

    private void CreateArray()
    {
      FArray = new int[10];
      for (int i = 0; i < 10; i++)
      {
        FArray[i] = i + 1;
      }
    }

    private void btnCreateNew_Click(object sender, EventArgs e)
    {
      // create report instance
      Report report = new Report();

      // register the array
      report.RegisterData(FArray, "Array");

      // design the report
      report.Design();

      // free resources used by report
      report.Dispose();
    }

    private void btnRunExisting_Click(object sender, EventArgs e)
    {
      // create report instance
      Report report = new Report();

      // load the existing report
      report.Load(@"..\..\report.frx");

      // register the array
      report.RegisterData(FArray, "Array");

      // run the report
      report.Show();

      // free resources used by report
      report.Dispose();
    }
  }
}